src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpUseImplicitTypeDiagnosticAnalyzer.cs

```diff
@@ -107,9 +107,18 @@ protected override bool TryAnalyzeVariableDeclaration(TypeSyntax typeName, Seman
                 }
 
                 var variable = variableDeclaration.Variables.Single();
-                if (AssignmentSupportsStylePreference(
-                        variable.Identifier, typeName, variable.Initializer.Value,
-                        semanticModel, optionSet, cancellationToken))
+                var initializer = variable.Initializer.Value;
+
+                // Discourage using "var pointer = stackalloc", after introducing C# 7.2, as it might be confusing to users expecting a Span<>
+                // Check descendant nodes as well, because Span-creating stackallocs can be nested in other nodes (like conditional operators and casts).
```

Comment:
that seems odd... what if i had something like ```Type t = foo(... stackalloc ...);```

This would still be a location where converting to 'var' woudl be ok, right?

Reply:
what if the stackalloc is inside a lambda that is on the right hand side?  it seems weird that this would disable this feature at this level.

Reply:
Such syntax is disabled. Please check #22046

Reply:
You can't have a stackalloc inside lambda on the RHS?

Reply:
I'll test that as well.